react/components/Drawer/index.js

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '../Button'
+import Spinner from '../Spinner'
+import ArrowBack from '../icon/ArrowBack'
+import config from 'vtex-tachyons/config.json'
+import Close from '../icon/Close'
+
+const ANIMATION_DURATION = 300
+
+class DrawerBox extends Component {
+  constructor(props) {
+    super(props)
+    this.overlay = React.createRef()
+    this.handleClick = this.handleClick.bind(this)
+    this.handleEscape = this.handleEscape.bind(this)
+    this.handleClose = this.handleClose.bind(this)
+  }
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClick, false)
+    document.addEventListener('keydown', this.handleEscape, false)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClick, false)
+    document.removeEventListener('keydown', this.handleEscape, false)
+  }
+
+  handleEscape({ key }) {
+    if (key === 'Escape') {
+      this.handleClose()
+    }
+  }
+
+  handleClick(e) {
+    if (this.overlay.current.contains(e.target)) {
+      // The click was inside the drawer element
+      return
+    }
+    this.handleClose()
+  }
+
+  handleClose() {
+    // The click was in the overlay area
+    const { isOpen, onClose } = this.props
+    if (isOpen) { onClose() }
+  }
+
+  render() {
+    return (
+      <div ref={this.overlay} className="flex flex-column items-stretch h-100">
+        { this.props.children }
+      </div>
+    )
+  }
+}
+
+class Drawer extends Component {
+  state = {
+    isVisible: false,
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.isOpen !== prevProps.isOpen) {
+      if (this.props.isOpen) {
+        this.setState({ isVisible: true })
+      } else {
+        setTimeout(() => {
```

Comment:
```suggestion
        this.hideTimeout = setTimeout(() => {
```

E no unmount chamar um `clearTimeout(this.hideTimeout)`